File: templates/verticle/src/main/resources/main_verticle.js
{{#containsDep dependencies "io.vertx" "vertx-web"}}
var Router = require("vertx-web-js/router");
{{/containsDep}}

// your code goes here...

{{#containsDep dependencies "io.vertx" "vertx-web"}}
var router = Router.router(vertx);

router.get("/").handler(function (ctx) {
  ctx.response()
    .putHeader("content-type", "text/plain")
    .end("Hello from Vert.x!");
});
{{/containsDep}}

vertx.createHttpServer().requestHandler({{#containsDep dependencies "io.vertx" "vertx-web"}}router{{else}}function (req) {
  req.response()
    .putHeader("content-type", "text/plain")
    .end("Hello from Vert.x!")
}{{/containsDep}}).listen(8080, function (res, res_err) {
  if (res_err != null) {
    res_err.printStackTrace();
  } else {
    console.log("Server listening at: http://localhost:8080/");
  }
});

File: templates/verticle/src/main/resources/main_verticle.rb
{{#containsDep dependencies "io.vertx" "vertx-web"}}
require 'vertx-web/router'
{{/containsDep}}

# your code goes here...

{{#containsDep dependencies "io.vertx" "vertx-web"}}
router = VertxWeb::Router.router($vertx)

router.get("/").handler() { |ctx|
  ctx.response()
    .putHeader("content-type", "text/plain")
    .end("Hello from Vert.x!")
}
{{/containsDep}}

http_server = $vertx.create_http_server()

http_server.request_handler({{#containsDep dependencies "io.vertx" "vertx-web"}}&router.method(:accept)){{else}}) { |req|
  req.response()
    .putHeader("content-type", "text/plain")
    .end("Hello from Vert.x!")
}{{/containsDep}}

http_server.listen(8080) { |res_err,res|
  if (res_err != nil)
    res_err.print_stack_trace()
  else
    puts "Server listening at: http://localhost:8080/"
  end
}
